python/cudf/cudf/core/index.py

```diff
@@ -2752,6 +2763,220 @@ def categories(self):
         return self._values.cat().categories
 
 
+def interval_range(
+    start: int = None,
+    end: int = None,
+    periods: int = None,
+    freq: int = None,
+    name: str = None,
+    closed: str = "right",
+) -> "IntervalIndex":
+    """
+    Returns a fixed frequency IntervalIndex.
+
+    Parameters
+    ----------
+    start : numeric, default None
+        Left bound for generating intervals.
+    end : numeric , default None
+        Right bound for generating intervals.
+    periods : int, default None
+        Number of periods to generate
+    freq : numeric, default None
+        The length of each interval. Must be consistent
+        with the type of start and end
+    name : str, default None
+        Name of the resulting IntervalIndex.
+    closed : {"left", "right", "both", "neither"}, default "right"
+        Whether the intervals are closed on the left-side, right-side,
+        both or neither.
+
+    Returns
+    -------
+    IntervalIndex
+
+    Examples
+    --------
+    >>> import cudf
+    >>> import pandas as pd
+    >>> cudf.interval_range(start=0,end=5)
+    IntervalIndex([(0, 0], (1, 1], (2, 2], (3, 3], (4, 4], (5, 5]],
+    ...closed='right',dtype='interval')
+    >>> cudf.interval_range(start=0,end=10, freq=2,closed='left')
+    IntervalIndex([[0, 2), [2, 4), [4, 6), [6, 8), [8, 10)],
+    ...closed='left',dtype='interval')
+    >>> cudf.interval_range(start=0,end=10, periods=3,closed='left')
+    ...IntervalIndex([[0.0, 3.3333333333333335),
+            [3.3333333333333335, 6.666666666666667),
+            [6.666666666666667, 10.0)],
+            closed='left',
+            dtype='interval')
+    """
+    if freq and periods and start and end:
+        raise ValueError(
+            "Of the four parameters: start, end, periods, and "
+            "freq, exactly three must be specified"
+        )
+    if isinstance(start or end, dt.datetime) or isinstance(
+        start or end, dt.timedelta
+    ):
+        raise NotImplementedError(
+            "Datetime-like intervals are not yet supported"
+        )
+    if start and isinstance(start, str):
+        raise ValueError(f"start must be numeric, got {start}")
+    if end and isinstance(end, str):
+        raise ValueError(f"end must be numeric, got {start}")
+    elif periods and not freq:
+        # if statement for mypy to pass
+        if end is not None and start is not None:
+            freq_step = ((end) - start) / periods
+            freq_step = DeviceScalar(freq_step, dtype="float64")
```

Comment:
Is it possible to use a `cudf.Scalar` here instead of a `DeviceScalar` explicitly, and make it so that `device_value` gets called inside `sequence`?

Reply:
Sure, I'll try this out!

Reply:
Thanks Brandon, that worked 😄